utils/manage_events.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+
+import logging
+import json
+from typing import Dict, List, Optional
+
+from cortx.utils.message_bus import MessageBus, MessageConsumer
+from ha.core.event_manager.event_manager import EventManager
+from ha.core.event_manager.subscribe_event import SubscribeEvent
+
+COMPONENT_ID = 'hare'
+
+class EventListener():
+    """This is listener for event manager
+
+    Typical use case of this class is as follows
+    subscribe(event list) - subscribe to events for listening, returns topic
+                            which can be used for listening
+    _listen(topic) - reading messages received on above topic
+    unsubscribe(event list) - unsubscribe as we dont want to receive these
+                              events anymore
+    get_message_type() - returns topic just in case user is not maintaining
+                         output of 'subscribe'
+    Note: 1. There is only 1 topic(message type, queue) for given component.
+          2. When component subcribes for event for the 1st time topic is
+             created and returned to user.
+          3. This topic will be used by user to listen for events.
+          4. After event is received in queue, user will read it, process it
+             and acknowledge it
+
+    """
+
+    event_manager = None
+    consumer = None
+
+    def __init__(self):
+        logging.debug('Inside EventListener')
+        self.event_manager = EventManager.get_instance()
+
+    def subscribe(self, event_list: List[SubscribeEvent]) -> str:
```

Comment:
I really doubt that `subscribe`/`unsubscribe` methods are really needed.

Internally, your EventListener can potentially use them, but I see no point to expose these two methods to the client. The client just wants to start receiving the events sent to hare.

Can this listener class check somewhere at the very beginning if hare is registered to receive such events and if not, automatically subscribe on them?

Reply:
@knekrasov  Client needs to provide list of events which it wants to receive(subscribe).
How client should communicate them if we remove subscribe/unsubscribe?

Reply:
The client can provide this list when creating EventListener (i.e. as `__init__` parameter).

And one more point: as far as I understand, subscribe/unsubscribe is cluster-wise action. In other words, they either subscribe ALL hare instances in the cluster to an event or the unsubscribe ALL hare instances. Is it correct?

If this is correct, then unsubscribing must be done with care: listening is done at every particular node and listen is node-specific. If one of the nodes stops listening, it should not unsubscribe because it is possible that there are other hare instances that are still willing to receive notifications.

Reply:
Done